### ISSUE TITLE
The impact of iota in the information criteria

### ISSUE THREAD

**Issue by TimVeenboer** [dsb-ifi/dHT#2]
Hi,

Great work on the paper, really nice implementation! I'm trying to apply the tokenization for a project of mine, however one of the parameters in the code is a bit unclear for me: https://github.com/dsb-ifi/dHT/blob/main/dht/tok/infocrit.py#L14-L15. 

The iota is stated to be for numerical stability, but I observe vastly different tokenizations when iota is changed only slightly. It seems to function as a penalty on area, where n_tokens increases with iota where a difference of 1e-3 around iota=1 seems to cause a doubling of the tokens produced by the tokenizer. See this image for reference on 4k res: 

<img width="2212" height="1266" alt="Image" src="https://github.com/user-attachments/assets/9886e92d-a1b9-43ed-87a3-9fba89b40fa6" />

What would be your guidance on the iota parameter? Which value were the pretrained models tuned with?

Many thanks & kind regards,
Tim

**Comment by mariuaas** [comment]
Hi Tim,

Sorry for the late response, and thanks for raising this. Your interpretation is correct: `iota` adds some additional behaviour in addition to acting as a  numerical-stability constant in the current implementation.

In [gaussian_2nll](https://github.com/dsb-ifi/dHT/blob/main/dht/tok/infocrit.py), the relevant term is:

```python
mV * log(prod(2 * pi * s2) + iota)
```

Here, `mV` is subsequently normalized by the image area. Changing `iota` therefore directly changes the model-fit term used to select among levels of the hierarchy. Since this contribution is weighted by region area, increasing `iota` penalizes larger merged regions more strongly and tends to favour finer tokenizations.

The difference you observe around `iota=1` is consistent with what we would expect. Selection is fundamentally discrete: if many regions are close to the information-criterion boundary, a small change in their scores can switch many decisions simultaneously. These changes may also propagate through the hierarchy, producing a large jump in the final token count. This can be particularly pronounced at high resolutions.

We should therefore distinguish the original numerical-stability motivation from the parameter’s effective role. Setting \(\iota=0\) can cause numerical instability, but positive values are not behaviourally neutral: in practice, `iota` also acts as a granularity control, or prior.

The public code currently uses `iota=5.0` as the default. However, `iota` is a runtime configuration rather than part of the checkpoint state, and the public repository and paper do not establish that every reported pretrained model was trained or tuned using exactly this value. We need to verify the original training configurations before stating that more strongly. 

For now, I would recommend treating `iota` as a resolution- and data-dependent granularity hyperparameter, and reporting it alongside token counts, rather than treating it solely as an implementation epsilon.
